http/handler.go

```diff
@@ -58,7 +58,7 @@ func Handler(core *vault.Core) http.Handler {
 	mux.Handle("/v1/sys/wrapping/unwrap", handleRequestForwarding(core, handleLogical(core, false, wrappingVerificationFunc)))
 	mux.Handle("/v1/sys/capabilities-self", handleRequestForwarding(core, handleLogical(core, true, nil)))
 	mux.Handle("/v1/sys/", handleRequestForwarding(core, handleLogical(core, true, nil)))
-	mux.Handle("/v1/", handleRequestForwarding(core, handleLogical(core, false, nil)))
+	mux.Handle("/v1/", handleCORS(core, handleRequestForwarding(core, handleLogical(core, false, nil))))
```

Comment:
We talked about this internally. The problem is defining which sys functions "make sense". A proper web UI would allow you -- if you are properly authorized -- to do nearly anything within Vault. It would be odd UX to say "hey, you have properly authenticated and are authorized but we've arbitrarily decided that this function requires `curl` instead of a browser".

Maybe you're concerned about e.g. browser extensions being able to do bad things, but those are both up to the user to accept and feels like prematurely optimizing. In the end I'm not sure this really benefits security but does make the UX feel disjoint.